decentraland/unity-explorer
# fix: PerformanceAnalyticsSystem reduce json allocations

## What does this PR change?

Json instantiation causes big allocation and GC spikes, IJsonObjectBuilder is introduced to solve this problem. It repools JsonObject and JsonElement for string values. 

The problem is not fully solved since JsonElement is immutable and heap allocated. It means that we could reassign values only via reflection that could be worse in performance then the current option. 

Caching for float values is not implemented due it requires special approach with objects comparing in dictionary (float precision problem) and could cause overusing memory without benefits

![image](https://github.com/user-attachments/assets/7761e49b-6a51-4b6c-996f-0155560889d0)

## How to test the changes?

1. Launch the explorer
2. Play Happy path, ensure that performance analytic's events work fine. 
3. Ensure Intel Mac also runs

Events to check 
- MOVE_TO_PARCEL - just move between different parcels;
- TIME_SPENT_IN_WORLD - spend some time in olavra.dcl.eth and then jump to different one or to Genesis;
- USED_EMOTE - play some emote;
-  MESSAGE_SENT - send a chat message;
- CHAT_BUBBLE_SWITCHED - turn chat bubble on and off;
- PASSPORT_OPENED - open/close some player passport, self passport;
- JUMP_IN - use different jump-in's - from the map; from carousel;

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
